components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/impl/CachedBackedSecretDAO.java

```diff
@@ -87,7 +87,6 @@ public Secret getSecretById(String secretId, int tenantId) throws SecretManageme
                 String message = String.format("Cache hit for secret by it's id. Secret id: %s", secretId);
                 log.debug(message);
             }
-            return secret;
```

Comment:
A similar change can be done to the `getSecretByName` method as well.

Reply:
Fixed